### PR TITLE
[pallet-revive] implement the call data size API

### PR DESCRIPTION
**Pull request by xermicus**
This PR adds an API method to query the contract call data input size.

Part of #6770

### CODE REVIEW

**Comment by xermicus** [comment]
bot fmt

File: substrate/frame/revive/src/tests.rs [hunk 1]
```diff
@@ -4342,6 +4342,28 @@ fn create1_with_value_works() {
...
+
+	ExtBuilder::default().existential_deposit(100).build().execute_with(|| {
+		let _ = <Test as Config>::Currency::set_balance(&ALICE, 1_000_000);
+
+		// Create fixture: Constructor does nothing
+		let Contract { addr, .. } =
+			builder::bare_instantiate(Code::Upload(code)).build_and_unwrap_contract();
+
+		// Call the contract: It echoes back the value returned by the call data size API.
+		let received = builder::bare_call(addr).build().result.unwrap();
```
Comment (athei) [inline substrate/frame/revive/src/tests.rs:4357]:
```suggestion
		let received = builder::bare_call(addr).build_and_unwrap_result();
```

**Review by athei — APPROVED** [review]

**Review by pgherveou — APPROVED** [review]

File: substrate/frame/revive/src/wasm/runtime.rs [hunk 2]
```diff
@@ -1299,6 +1302,22 @@ pub mod env {
 		)
 	}
 
+	/// Returns the total size of the contract call input data.
+	/// See [`pallet_revive_uapi::HostFn:: call_data_size `].
```
Comment (athei) [inline substrate/frame/revive/src/wasm/runtime.rs:1306]:
```suggestion
	/// See [`pallet_revive_uapi::HostFn::call_data_size `].
```

**Pull request merged**